independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -239,17 +241,20 @@ private static ScopeInfo initStereotypeScope(List<StereotypeInfo> stereotypes, A
         if (stereotypes.isEmpty()) {
             return null;
         }
-        ScopeInfo stereotypeScope = null;
+        final Set<ScopeInfo> stereotypeScopes = new HashSet<>();
         for (StereotypeInfo stereotype : stereotypes) {
-            if (stereotype.getDefaultScope() != stereotypeScope) {
-                if (stereotypeScope == null) {
-                    stereotypeScope = stereotype.getDefaultScope();
-                } else {
-                    throw new IllegalStateException("All stereotypes must specify the same scope or the bean must declare a scope: " + target);
-                }
-            }
+            stereotypeScopes.addAll(stereotype.getDefaultScopes());
+        }
+        switch (stereotypeScopes.size()) {
+            case 0:
+                return null;
+            case 1:
+                return stereotypeScopes.iterator().next();
+            default:
+                throw new DefinitionException("All stereotypes must specify the same scope or the bean must declare a scope: "
```

Comment:
I think that we should rather fail in the `BeanDeployment.findStereotypes()` method when multiple scopes are declared on a stereotype annotation. There is no need to wait until a bean attempts to use it.

Reply:
Done.